### PR TITLE
[DOC-1498] CI tutorial - Build and test Java

### PR DESCRIPTION
**Pull request by aimurphy**
Revise Java app tutorial.

https://6446f04ada3d2918087a9aa6--harness-developer.netlify.app/tutorials/ci-pipelines/build/java

### CODE REVIEW

**Review by michaelcretzman — DISMISSED** [review]
my only suggestion is to be consistent with articles in titles. Some use "a" and "the" while others are zero article.

File: tutorials/ci-pipelines/build/ci-java-http-server.md [hunk 1]
```diff
@@ -20,126 +21,245 @@ import CISignupTip from '/tutorials/shared/ci-signup-tip.md';
...
+7. On the Modules page, select **Continuous Integration**, and then select **Go to Module**.
 
-### Docker Hub connector
+If this is your first project with CI, the CI pipeline wizard starts after you select **Go to Module**. You'll need to exit the wizard to create the Docker Hub connector.
 
-You need a [Docker Hub](https://hub.docker.com/) connector. This connector is used to authenticate and publish the Java HTTP server image to your Docker Hub account.
+</details>
 
-If you have not created a Docker Hub connector yet, follow these steps.
+### Create a GitHub connector
```
Comment (aimurphy) [inline tutorials/ci-pipelines/build/ci-java-http-server.md:48]:
```suggestion
### Create the GitHub connector
```

File: tutorials/ci-pipelines/build/ci-java-http-server.md [hunk 2]
```diff
@@ -20,126 +21,245 @@ import CISignupTip from '/tutorials/shared/ci-signup-tip.md';
...
                     shell: Sh
-                    command: echo "Welcome to Harness CI"
+                    command: |-
+                      echo "Welcome to Harness CI"
+                      mvn -B package --file pom.xml
+```
+
+</details>
+
+### Understand the build infrastructure
```
Comment (aimurphy) [inline tutorials/ci-pipelines/build/ci-java-http-server.md:144]:
```suggestion
### Understand build infrastructure
```

File: tutorials/ci-pipelines/build/ci-java-http-server.md [hunk 3]
```diff
@@ -20,126 +21,245 @@ import CISignupTip from '/tutorials/shared/ci-signup-tip.md';
...
+
+In contrast, if you choose to [use a Kubernetes cluster build infrastructure](/docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/set-up-a-kubernetes-cluster-build-infrastructure) and your pipeline requires a tool that is not already available in the cluster, you can configure your pipeline to load those prerequisite tools when the build runs. There are several ways to do this in Harness CI, including:
+
+* [Background steps](/docs/continuous-integration/use-ci/manage-dependencies/dependency-mgmt-strategies) for running dependent services.
+* [Plugin steps](/docs/continuous-integration/use-ci/use-drone-plugins/explore-ci-plugins) to run templated scripts, such as GitHub Actions, BitBucket Integrations, or any Drone plugin.
+* [Various caching options](/docs/continuous-integration/use-ci/caching-ci-data/share-ci-data-across-steps-and-stages) to load dependency caches.
+* [Run steps](/docs/category/run-scripts) for running all manner of scripts and commands.
+
+## Add a pipeline variable
```
Comment (aimurphy) [inline tutorials/ci-pipelines/build/ci-java-http-server.md:172]:
```suggestion
## Use variables
```

File: tutorials/ci-pipelines/build/ci-java-http-server.md [hunk 4]
```diff
@@ -153,21 +273,110 @@ Replace the sample `stages` section with the following `variables` and `stages`
...
+You could also use `preCommand` to prepare the test environment. For example, you could supply commands to [install Gradle](https://gradle.org/install/).
+
+:::
+
+```mdx-code-block
+  </TabItem>
+</Tabs>
+```
+
+## Build and push an image to Docker Hub
```
Comment (aimurphy) [inline tutorials/ci-pipelines/build/ci-java-http-server.md:293]:
```suggestion
## Build and push to Docker Hub
```

File: tutorials/ci-pipelines/build/ci-java-http-server.md [hunk 5]
```diff
@@ -20,126 +21,245 @@ import CISignupTip from '/tutorials/shared/ci-signup-tip.md';
...
-          execution:
-            steps:
+```
+
+```mdx-code-block
+  </TabItem>
+</Tabs>
+```
+
+## Run initial tests
```
Comment (aimurphy) [inline tutorials/ci-pipelines/build/ci-java-http-server.md:212]:
```suggestion
## Run the initial tests
```

**Pull request merged**